docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -25,107 +26,32 @@ command:
+1. Configure a shared secret for internal communication as described in
+   the preceding section.
 
-       http-server.http.enabled=false
+2. Enable automatic certificate creation and trust setup in
+   ``etc/config.properties``:
 
-   .. warning::
+   .. code-block:: properties
 
-       You can enable HTTPS, while leaving HTTP enabled. In most cases this is a
-       security hole. If you are certain you want to use this configuration, you
-       should consider using an firewall to limit access to the HTTP endpoint to
-       only those hosts that should be allowed to use it.
+     http-server.https.automatic-shared-secret=automatic-secret
 
-2. Configure the cluster to communicate using the fully qualified domain name (fqdn)
-   of the cluster nodes. This can be done in either of the following ways:
+3. Change the discovery uri to HTTPS in ``etc/config.properties``:
 
-   - If the DNS service is configured properly, we can just let the nodes
-     introduce themselves to the coordinator using the hostname taken from
-     the system configuration (``hostname --fqdn``)
-
-     .. code-block:: text
-
-         node.internal-address-source=FQDN
-
-   - It is also possible to specify each node's fully-qualified hostname manually.
-     This is different for every host. Hosts should be in the same domain to
-     make it easy to create the correct SSL/TLS certificates.
-     e.g.: ``coordinator.example.com``, ``worker1.example.com``, ``worker2.example.com``.
-
-     .. code-block:: text
-
-         node.internal-address=<node fqdn>
-
-
-3. Generate a Java Keystore File. Every Trino node must be able to connect to
-   any other node within the same cluster. It is possible to create unique
-   certificates for every node using the fully-qualified hostname of each host,
-   create a keystore that contains all the public keys for all of the hosts,
-   and specify it for the client (see step #8 below). In most cases it is
-   simpler to use a wildcard in the certificate as shown below.
-
-   .. code-block:: text
-
-       keytool -genkeypair -alias example.com -keyalg RSA -keystore keystore.jks
-       Enter keystore password:
-       Re-enter new password:
-       What is your first and last name?
-         [Unknown]:  *.example.com
-       What is the name of your organizational unit?
-         [Unknown]:
-       What is the name of your organization?
-         [Unknown]:
-       What is the name of your City or Locality?
-         [Unknown]:
-       What is the name of your State or Province?
-         [Unknown]:
-       What is the two-letter country code for this unit?
-         [Unknown]:
-       Is CN=*.example.com, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown correct?
-         [no]:  yes
-
-   .. Note: Replace `example.com` with the appropriate domain.
-
-4. Distribute the Java Keystore File across the Trino cluster.
-
-5. Enable the HTTPS endpoint.
-
-   .. code-block:: text
-
-       http-server.https.enabled=true
```

Comment:
We need the first two (enabled and port)

The keystore (and maybe truststore) is needed on the coordinator for TLS for clients.

Reply:
I added that back in to some degree .. but mostly I am linking to the TLS page to avoid repeating stuff. It all gets a bit hairy when people use an LB with TLS termination and whatever some people want to do..